packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -62,71 +61,36 @@ const DummyDisabledButton = styled.div`
 
 const SignupFlowPlanFeatureActionButton = ( {
 	planSlug,
-	planTitle,
-	priceString,
 	isStuck,
-	isLargeCurrency,
 	hasFreeTrialPlan,
 	onCtaClick,
 	onFreeTrialCtaClick,
-	planActionOverrides,
+	text,
+	freeTrialText,
+	postButtonText,
+	status,
 }: {
 	planSlug: PlanSlug;
-	planTitle: TranslateResult;
-	priceString: string | null;
 	isStuck: boolean;
-	isLargeCurrency: boolean;
 	hasFreeTrialPlan: boolean;
 	onCtaClick: () => void;
 	onFreeTrialCtaClick: () => void;
-	planActionOverrides?: PlanActionOverrides;
+	text: string;
+	freeTrialText?: string;
+	postButtonText: string | null;
+	status: string;
 } ) => {
-	const translate = useTranslate();
-	const busy =
-		isFreePlan( planSlug ) && planActionOverrides?.loggedInFreePlan?.status === 'blocked';
-	const postButtonText =
-		isBusinessPlan( planSlug ) && planActionOverrides?.trialAlreadyUsed?.postButtonText;
-
-	let btnText = translate( 'Get %(plan)s', {
-		args: {
-			plan: planTitle,
-		},
-	} );
-
-	if ( isFreePlan( planSlug ) ) {
-		btnText = translate( 'Start with Free' );
-	} else if ( isStuck && ! isLargeCurrency ) {
-		btnText = translate( 'Get %(plan)s – %(priceString)s', {
-			args: {
-				plan: planTitle,
-				priceString: priceString ?? '',
-			},
-			comment:
-				'%(plan)s is the name of the plan and %(priceString)s is the full price including the currency. Eg: Get Premium - $10',
-		} );
-	} else if ( isStuck && isLargeCurrency ) {
-		btnText = translate( 'Get %(plan)s {{span}}%(priceString)s{{/span}}', {
-			args: {
-				plan: planTitle,
-				priceString: priceString ?? '',
-			},
-			comment:
-				'%(plan)s is the name of the plan and %(priceString)s is the full price including the currency. Eg: Get Premium - $10',
-			components: {
-				span: <span className="plan-features-2023-grid__actions-signup-plan-text" />,
-			},
-		} );
-	}
+	const busy = isFreePlan( planSlug ) && status === 'blocked';
 
 	if ( hasFreeTrialPlan ) {
 		return (
 			<div className="plan-features-2023-grid__multiple-actions-container">
 				<PlanButton planSlug={ planSlug } onClick={ () => onFreeTrialCtaClick() } busy={ busy }>
-					{ translate( 'Try for free' ) }
+					{ freeTrialText }
 				</PlanButton>
 				{ ! isStuck && ( // along side with the free trial CTA, we also provide an option for purchasing the plan directly here
 					<PlanButton planSlug={ planSlug } onClick={ onCtaClick } borderless>
-						{ btnText }
+						{ text }
 					</PlanButton>
```

Comment:
So this can be the end/ultimate form, along with `postButtonText` at the end of it. For the free-trial actions, we'd have the free-trial-CTA as the "primary" action and the other as "secondary". For everything else, we'd not be providing a "secondary" action instead.

Reply:
Good callout about the free trial text. I'll make sure it ends up in our abstracted action button component